test/hello-query.ts

```diff
@@ -26,4 +36,64 @@ describe('hello query test', function () {
     const content = response.data.data.hello[0].content
     expect(content).to.be.equal('Hello world!')
   })
+
+  const test_input = {
+    data: {
+      name: 'test_name',
+      email: 'test_email1',
+      password: 'test_password1',
+      birthDate: '01-01-1900',
+    },
+  }
+
+  it('should return a user as response', async function () {
+    const response = await axios.post('http://localhost:4000', {
+      query: `
+        mutation CreateUser($data: UserInput!) {
+          createUser(data: $data) {
+            birthDate
+            email
+            id
+            name
+          }
+        }
+      `,
+      variables: {
+        data: {
+          name: test_input.data.name,
+          email: test_input.data.email,
+          password: test_input.data.password,
+          birthDate: test_input.data.birthDate,
+        },
+      },
+    })
+
+    console.log(response.data)
+    expect(response.data.data).not.to.be.null
+    expect(response.data.data.createUser.id).not.to.be.null
+    createdUserId = response.data.data.createUser.id
+    expect(response.data.data.createUser).not.to.have.property('password')
+    expect(response.data.data.createUser.birthDate).to.be.equal(
+      test_input.data.birthDate
+    )
+    expect(response.data.data.createUser.email).to.be.equal(
+      test_input.data.email
+    )
+    expect(response.data.data.createUser.name).to.be.equal(test_input.data.name)
+  })
```

Comment:
Provavelmente vc consegue trocar todas essas afirmacões pela `to.be.deep.eq`, pq ele ja checa se todos os campos e valores estão de acordo com o segundo objeto que vc envia, e inclusive ele falha se tiver faltando/sobrando campo. Para testar responses do GraphQL ele é muito bom.